tests/sympc/protocol/falcon/falcon_test.py

```diff
@@ -242,3 +243,52 @@ def test_prime_mul_private(get_clients, security):
     expected_res = prime_op(secret1, secret2)
 
     assert (result.reconstruct(decode=False) == expected_res).all()
+
+
+@pytest.mark.parametrize("security", ["semi-honest", "malicious"])
+def test_select_shares(get_clients, security) -> None:
+    parties = get_clients(3)
+    falcon = Falcon(security)
+    session = Session(parties=parties, protocol=falcon)
+    SessionManager.setup_mpc(session)
+    sh = torch.tensor([[1, 0], [0, 1]], dtype=torch.bool)
+    shares = [sh, sh, sh]
+    ptr_lst = ReplicatedSharedTensor.distribute_shares(shares, session, ring_size=2)
+    b = MPCTensor(shares=ptr_lst, session=session, shape=sh.shape)
+
+    x_val = torch.tensor([[1, 2], [3, 4]])
+    y_val = torch.tensor([[5, 6], [7, 8]])
+    x = MPCTensor(secret=x_val, session=session)
+    y = MPCTensor(secret=y_val, session=session)
+
+    z = Falcon.select_shares(x, y, b)
+
+    tensor_type = get_type_from_ring(session.ring_size)
+    bit_tensor = b.reconstruct(decode=False).type(tensor_type)
+    result = (x.reconstruct() * (bit_tensor ^ 1)) + (y.reconstruct() * bit_tensor)
+
+    assert (result == z.reconstruct()).all()
+
+
+def test_select_shares_exception_ring(get_clients) -> None:
+    parties = get_clients(3)
+    falcon = Falcon()
+    session = Session(parties=parties, protocol=falcon)
```

Comment:
MAYBE: Could you also manually specify here the ring (it might be easier for someone that reads the tests) - I know that the default is != 2 ,but someone new might not know

Reply:
Yes crct george , I will modify it :+1:

Reply:
Done